translation/transport.go

```diff
@@ -65,24 +69,64 @@ func ConfigHandler(c *Options) {
 		Methods(http.MethodDelete, http.MethodPut, http.MethodPost)
 }
 
-/* Request Decoding */
+const (
+	PartnerIdHeader               = "X-Xmidt-Partner-Id"
+)
+// getPartnerIDs returns the array that represents the partner-ids that were
+// passed in as headers.  This function handles multiple duplicate headers.
+func getPartnerIDs(h http.Header) []string {
+	headers, ok := h[PartnerIdHeader]
+	if !ok {
+		return nil
+	}
 
+	var partners []string
+
+	for _, value := range headers {
+		fields := strings.Split(value, ",")
+		for i := 0; i < len(fields); i++ {
+			fields[i] = strings.TrimSpace(fields[i])
+		}
+		partners = append(partners, fields...)
+	}
+
+	return partners
+}
+
+
+/* Request Decoding */
 func decodeRequest(ctx context.Context, r *http.Request) (decodedRequest interface{}, err error) {
 	var (
 		payload []byte
 		wrpMsg  *wrp.Message
 	)
-
 	if payload, err = requestPayload(r); err == nil {
 		var tid = ctx.Value(common.ContextKeyRequestTID).(string)
-		if wrpMsg, err = wrap(payload, tid, mux.Vars(r)); err == nil {
+		var partnerIDs []string
+		auth, ok := bascule.FromContext(ctx)
+		//if no token
+		if !ok {
+			partnerIDs = getPartnerIDs(r.Header)
+		}
+		tokenType := auth.Token.Type()
+		//if not jwt type
+		if tokenType != "jwt" {
+			partnerIDs = getPartnerIDs(r.Header)
+		}
+		if tokenType == "jwt" {
+			partnerIDs, ok = auth.Token.Attributes().GetStringSlice(basculechecks.PartnerKey)
+			//if no partner ids
+			if !ok {
+				partnerIDs = getPartnerIDs(r.Header)
+			}
+		}
```

Comment:
I would suggest making this its own function (with context as the parameter) so that you can return when you get the partnerIDs you want; otherwise, `else`s should be added so that we don't continue through code after hitting a negative case.

Reply:
do we just want to return from the func if it can't get partnerIDs or if the token type isn't jwt?